python/mxnet/gluon/trainer.py

```diff
@@ -191,6 +224,8 @@ def step(self, batch_size, ignore_stale_grad=False):
         """
         if not self._kv_initialized:
             self._init_kvstore()
+        if self._params_to_init:
```

Comment:
I don't quite understand this. If there are uninitialized parameters, wouldn't step fail?